v2/file-format-conversion/src/main/java/com/google/cloud/teleport/v2/templates/FileFormatConversionFactory.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (C) 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.teleport.v2.templates;
+
+import static org.apache.beam.vendor.guava.v20_0.com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.auto.value.AutoValue;
+import com.google.cloud.teleport.v2.templates.FileFormatConversion.FileFormatConversionOptions;
+import com.google.cloud.teleport.v2.templates.FileFormatConversion.ValidFileFormats;
+import com.google.cloud.teleport.v2.transforms.AvroConverters;
+import com.google.cloud.teleport.v2.transforms.CsvConverters;
+import com.google.cloud.teleport.v2.transforms.ParquetConverters;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.beam.sdk.coders.AvroCoder;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+import org.apache.beam.sdk.values.TupleTag;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Factory class for the {@link FileFormatConversion} class. */
+public class FileFormatConversionFactory {
+  /** The tag for the headers of the CSV if required. */
+  private static final TupleTag<String> CSV_HEADERS = new TupleTag<String>() {};
+
+  /** The tag for the lines of the CSV. */
+  private static final TupleTag<String> CSV_LINES = new TupleTag<String>() {};
+
+  /** Logger for class. */
+  private static final Logger LOG = LoggerFactory.getLogger(FileFormatConversionFactory.class);
+
+  /**
+   * The {@link FileFormat} class is a {@link PTransform} that reads an input file, converts it to a
+   * desired format and returns a {@link PDone}.
+   */
+  @AutoValue
+  public abstract static class FileFormat extends PTransform<PBegin, PDone> {
+
+    public static Builder newBuilder() {
+      return new AutoValue_FileFormatConversionFactory_FileFormat.Builder();
+    }
+
+    public abstract FileFormatConversionOptions options();
+
+    @Override
+    public PDone expand(PBegin input) {
+      String inputFileFormat = options().getInputFileFormat().toUpperCase();
+      String outputFileFormat = options().getOutputFileFormat().toUpperCase();
```

Comment:
This piece of code is repeated in the main pipeline class too. Can we avoid repeating this logic?